lib/plugins/aws/package/compile/events/kafka.js

```diff
@@ -112,6 +124,37 @@ class AwsCompileKafkaEvents {
       functionObj.events.forEach((event) => {
         if (!event.kafka) return;
 
+        if (!event.kafka.accessConfigurations || event.kafka.accessConfigurations.length === 0) {
+          throw new ServerlessError(
+            `You must provide at least one accessConfiguration for function: ${functionName}`,
+            'FUNCTION_KAFKA_ACCESS_CONFIGURATION_EMPTY'
+          );
+        }
+
+        const {
+          accessConfigurations: {
+            vpcSecurityGroup,
+            vpcSubnet,
+            clientCertificateTlsAuth,
+            serverRootCaCertificate,
+          },
+        } = event.kafka;
+
+        if ((vpcSecurityGroup && !vpcSubnet) || (vpcSubnet && !vpcSecurityGroup)) {
+          const missing = vpcSecurityGroup ? 'vpcSubnet' : 'vpcSecurityGroup';
+          throw new ServerlessError(
+            `You must specify at least one ${missing} accessConfiguration for function: ${functionName}`,
+            `FUNCTION_KAFKA_VPC_ACCESS_CONFIGURATION_MISSING_${missing.toUpperCase()}`
```

Comment:
Let's use static error code e.g. just `FUNCTION_KAFKA_VPC_ACCESS_CONFIGURATION_INVALID`, it will be easier to monitor these type of errors and distinction doesn't really bring much value here

Reply:
ok no probs. i'll still keep the "missing" dynamically to the message as this makes for a better developer experience

Reply:
agreed, message should be as easy for user to understand as possible 👍